Revise targeting docs (#570)
configcat/docs: website/docs/targeting/targeting-rule/targeting-rule-overview.mdx

<img alt="Add targeting rule" src="/docs/assets/targeting/targeting-rule/add-targeting-rule_192dpi.png" width="700" height="525" decoding="async" loading="lazy" />

## How does it work? - Anatomy of a Targeting Rule

A Targeting Rule consists of an _IF part_ and a _THEN part_.

<img alt="Targeting rule anatomy" src="/docs/assets/targeting/targeting-rule/targeting-rule-anatomy_192dpi.png" width="1084" height="433" decoding="async" loading="lazy" />

### IF part

The _IF part_ contains the conditions, which are logical expressions that evaluate to true or false.

The conditions are in an **AND** relationship, meaning that all of them must evaluate to true for the Targeting Rule to match.

The conditions can be added to the Targeting Rule on the Dashboard, using the **+AND** ("Add AND condition") button. There are three types of conditions:
- [User Condition](../user-condition) - A condition that is based on some property of the user.
- [Flag Condition (Prerequisite)](../flag-condition) - A condition that is based on the value of another feature flag.
- [Segment Condition](../segment-condition) - A condition that is based on a segment.

### THEN part

> The _THEN part_ contains either a value or Percentage Options that will be used to determine the result of the feature flag when the Targeting Rule matches.

Author: ```suggestion
The _THEN part_ determines the result of the feature flag when the Targeting Rule matches. It can be either a simple value or Percentage Options.
```

In our demo company (Whisker Co.) we have a new feature in our mobile app that notifies frequent shopper cat owners about the cat-friendly cafés in the neighborhood. 

#### Goal

> Since this feature is new, we want to make sure that only frequent shoppers and cat owners who have the right version of the app receive these notifications. The earlier versions of the app don't have this feature, so we want to make sure that only those who are running version 3.0.0 or higher receive the notifications.

Author: Tördeljük még.
